src/dynamic_programming/python/can_sum/src/can_sum.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Python Version: 3.12
+"""
+File: can_sum.py
+Author: Vlad421
+Date: 1/20/2024
+Description: Checks if array could produce given sum
+"""
+
+
+def can_sum_basic(number: int, arr: list[int]) -> bool:
+    """Basic brute force algorythm
+
+    Parameters:
+        arr: list[int] - List of items to sum
+        number:int - Target sum
+
+    Returns -> bool    
+
+    >>> can_sum_basic([2,1],1)
+    True """
+    if number < 0:
+        return False
+    elif number == 0:
+        return True
+
+    for i in arr:
+        reminder = number-i
+
+        return can_sum_basic(reminder, arr[1:]) or can_sum_basic(number, arr[1:])
+    return False
```

Comment:
I see now

    for i in arr:
   
does not make any sence - it's confusing

as this last return False
because you use
        return can_sum_basic(reminder, arr[1:]) or can_sum_basic(number, arr[1:])
function will return and nor 'for' loop, nor last ruturn will excecute

so you could write:
```
i=arr[0]
reminder = number-i
return can_sum_basic(reminder, arr[1:]) or can_sum_basic(number, arr[1:])
```

this will make code much better

Reply:
this not goon to work. second iteration will be next number  of array and you will check everything to [0] index of array

Reply:
Yes, I forgot the case when len(arr)==0, then iteration will not start, and you will gor to line 32 and return False, it a zero case

But the for here is not needed. So it's no problem to merge, but it's kind of mistake and code confusion

```
for i in range(1,10):
    j=some_function(i)*100
    return j
```
is totally equvalent to:
```
i=1
j=some_function(i)*100
return j
````

after retun loop will not run

I rewrote your code this way and it works:
```
    # for i in arr:
    #     reminder = number-i

    #     return can_sum_basic(reminder, arr[1:]) or can_sum_basic(number, arr[1:])
    
    # Zero case
    if len(arr)==0:
        return False

    return can_sum_basic(number-arr[0], arr[1:]) or can_sum_basic(number, arr[1:])
    # return False

```

Reply:
your solution is your solution. im not going to play wit it more. font like it - you can write your own solution.

Reply:
This is not about the solution or whose code is better, but as you wish